src/currentprocess.rs

```diff
@@ -111,6 +46,115 @@ impl Process {
+            #[cfg(feature = "test")]
+            Process::TestProcess(p) => p.vars.get(key).map(OsString::from),
+        }
+    }
+
+    pub(crate) fn args(&self) -> Box<dyn Iterator<Item = String> + '_> {
+        match self {
+            Process::OSProcess(_) => Box::new(env::args()),
+            #[cfg(feature = "test")]
+            Process::TestProcess(p) => Box::new(p.args.iter().cloned()),
+        }
+    }
+
+    pub(crate) fn args_os(&self) -> Box<dyn Iterator<Item = OsString> + '_> {
+        match self {
+            Process::OSProcess(_) => Box::new(env::args_os()),
+            #[cfg(feature = "test")]
+            Process::TestProcess(p) => Box::new(p.args.iter().map(OsString::from)),
+        }
+    }
+
+    pub(crate) fn stdin(&self) -> Box<dyn filesource::Stdin> {
+        match self {
+            Process::OSProcess(_) => Box::new(io::stdin()),
+            #[cfg(feature = "test")]
+            Process::TestProcess(p) => Box::new(filesource::TestStdin(p.stdin.clone())),
+        }
+    }
+
+    pub(crate) fn stdout(&self) -> Box<dyn filesource::Writer> {
+        match self {
+            Process::OSProcess(_) => Box::new(io::stdout()),
+            #[cfg(feature = "test")]
+            Process::TestProcess(p) => Box::new(filesource::TestWriter(p.stdout.clone())),
+        }
+    }
+
+    pub(crate) fn stderr(&self) -> Box<dyn filesource::Writer> {
+        match self {
+            Process::OSProcess(_) => Box::new(io::stderr()),
+            #[cfg(feature = "test")]
+            Process::TestProcess(p) => Box::new(filesource::TestWriter(p.stderr.clone())),
+        }
+    }
+
+    pub(crate) fn current_dir(&self) -> io::Result<PathBuf> {
+        match self {
+            Process::OSProcess(_) => env::current_dir(),
+            #[cfg(feature = "test")]
+            Process::TestProcess(p) => Ok(p.cwd.clone()),
+        }
+    }
+
+    #[cfg(test)]
+    fn id(&self) -> u64 {
+        match self {
+            Process::OSProcess(_) => std::process::id() as u64,
+            #[cfg(feature = "test")]
+            Process::TestProcess(p) => p.id,
+        }
+    }
+}
+
+impl home::env::Env for Process {
+    fn home_dir(&self) -> Option<PathBuf> {
+        match self {
+            Process::OSProcess(_) => self.var("HOME").ok().map(|v| v.into()),
+            #[cfg(feature = "test")]
+            Process::TestProcess(_) => home::env::OS_ENV.home_dir(),
+        }
+    }
+
+    fn current_dir(&self) -> Result<PathBuf, io::Error> {
+        match self {
+            Process::OSProcess(_) => self.current_dir(),
+            #[cfg(feature = "test")]
+            Process::TestProcess(_) => home::env::OS_ENV.current_dir(),
+        }
+    }
+
+    fn var_os(&self, key: &str) -> Option<OsString> {
+        match self {
+            Process::OSProcess(_) => self.var_os(key),
+            #[cfg(feature = "test")]
+            Process::TestProcess(_) => self.var_os(key),
+        }
+    }
```

Comment:
As I understand it, this delegates to the version of `.var_os()` in `Self`'s `impl` block. Is it necessary to create two versions of `.var_os()` though?

Reply:
Oh, it makes perfect sense if we want to minimize the imports being used.